error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -78,6 +78,24 @@ Mono<T> after(Optional<T> optional) {
     }
   }
 
+  /**
+   * Prefer a {@link Mono#justOrEmpty(Optional)} and {@link Mono#switchIfEmpty(Mono)} chain over
+   * more contrived alternatives.
+   */
+  static final class MonoFromOptionalSwitchIfEmpty<T> {
+    @BeforeTemplate
+    Mono<T> before(Optional<T> optional, Mono<T> alternative) {
+      return optional
+          .map(Refaster.<Function<T, Mono<T>>>anyOf(Mono::just, Mono::justOrEmpty))
```

Comment:
The `Mono::justOrEmpty` -> `Mono::just` case can be its own "precursor rule", following our approach to defining the smallest improvement per Refaster rule. Will push a proposal.

Reply:
Good one, thanks. I didn't see that initially.